HW-6 DI의존성 추가

jnugg0819 opened the pull request:
HW-6 Hilt를 이용한 DI추가

jnugg0819 committed:
HW-6 DI의존성 추가

- 모든 모듈 Hilt추가
- Frgagment에 ViewModel 생성 로직 부분 제거
- ViewModel, Fragment, Activity HIlt 적용하도록 어노테이션 추가

realizer12 commented on local/src/main/java/org/techtown/local/feature/di/LocalModule.kt:
```diff
@@ -0,0 +1,28 @@
+    @Provides
+    fun provideNewsDao(
+        @ApplicationContext context: Context
+    ) : ArticlesDao = AppDatabase.getInstance(context).articleDao()
```
상미님 지금 현재는 local database의 경우 내부 의존성이 생겨있는 상태이기때문에
외부 의존성으로바꿔 주는게 좋을거  같아엽
AppDatabas도  hilt 처리 해주세엽~!

jnugg0819 replied:
9dbe0b7abcb7244eea951f5685f5945a2c8070e5

realizer12 commented on presentation/src/main/java/org/techtown/presentation/feature/main/viewmodel/SavedViewModel.kt:
```diff
@@ -5,13 +5,16 @@ import androidx.lifecycle.MutableLiveData
 
-class SavedViewModel(
+@HiltViewModel
+class SavedViewModel @Inject constructor(
```
상미님 여기 동작 이상해서 살펴보니까
shouldRequestViewMore 이 필요없는 코드 같아엽
그리고 if (presentArticles.isNotEmpty()) 이거 false일때는 
list clear 를 해주어야 할것 같은데  확인점 

그리고 탭전환시 업데이트 안되는 문제때문에 getSavedArticleList 요놈은 fragment쪽
init에 들어가는게 좋을거 같아엽.

bottom nav사용시에는 fragment destory되어도 fragment용  viewmodel clear 가 안되는거 같아엽
이거는 좀더 알아봐야 할거 같은데
이문제대문에   탭왔다갔다했을때 ,   getSavedArticleList가 업데이트 안되는즁

jnugg0819 replied:
@realizer12 동훈님 이거 멘트 이상한데다 단거같은데 맞져?

jnugg0819 replied:
98c36908e5bcca32bcd179a54f30756f58373e48

realizer12 replied:
야스   맞아엽 ㅋㅋㅋ
상미님이 고친 부분 말한거 마즘  👍

jnugg0819 committed:
HW-6 DataBase도 의존성 추가

jnugg0819 committed:
HW-6 즐겨찾기 리스트 업데이트 안되던 부분 수정

- getSaved -> fetchSaved로변경 get을 쓰게되면 xml viewmodel 바인딩부분에서 해당 함수로 인식이됨
- 게시글 가져올때마다 clear해주는문 추가
